### PR TITLE
GODRIVER-1736 Fix deprecation notice formats

### PR DESCRIPTION
**Pull request by divjotarora**
I found these deprecations via `git grep "deprecated"`. After fixing the notices, I ran the `staticcheck` linter against https://play.golang.org/p/3jPWZqPMt0H and ensured that all of the usages are marked as deprecated.

If anyone wants to reproduce the results, run `go get honnef.co/go/tools/cmd/staticcheck` to install the linter and copy/paste the the code from the playground link into a local script.

I couldn't test deprecation warnings for struct fields due to https://github.com/dominikh/go-tools/issues/607. This means that the `AuthenticateToAnything` and `Deployment` Client options can't be tested because they don't have corresponding setter functions. To work around this for `FindOptions`, `FindOneOptions`, and `IndexOptions`, I deprecated both the struct field and the setter function.

### CODE REVIEW

File: mongo/index_options_builder.go [hunk 1]
```diff
@@ -10,125 +10,167 @@ import (
...
+// Name spcifies a value for the name option.
+//
+// Deprecated: Use the IndexOptions.SetName function in mongo/options instead.
 func (iob *IndexOptionsBuilder) Name(name string) *IndexOptionsBuilder {
 	iob.document = append(iob.document, bson.E{"name", name})
 	return iob
 }
 
-// Sparse is deprecated.
+// Sparse spcifies a value for the sparse option.
```
Comment (iwysiu) [inline mongo/index_options_builder.go:51]: all of the spcifies -> specifies

File: mongo/cursor.go [hunk 2]
```diff
@@ -269,8 +269,11 @@ func (c *Cursor) closeImplicitSession() {
 }
 
 // BatchCursorFromCursor returns a driver.BatchCursor for the given Cursor. If there is no underlying
-// driver.BatchCursor, nil is returned. This method is deprecated and does not have any stability guarantees. It may be
-// removed in the future.
+// driver.BatchCursor, nil is returned.
+//
+// Deprecated: This is an unstable function because the driver.BatchCursor type exists in the "x" package. Both this
```
Comment (iwysiu) [inline mongo/cursor.go:274]: nit: Neither this function nor the driver.BatchCursor type should be

File: mongo/options/findoptions.go [hunk 3]
```diff
@@ -357,8 +365,10 @@ type FindOneOptions struct {
 	// The number of documents to skip before selecting the document to be returned. The default value is 0.
 	Skip *int64
 
-	// If true, the cursor will not return a document more than once because of an intervening write operation. This
-	// option has been deprecated in MongoDB version 4.0. The default value is false.
+	// If true, the cursor will not return a document more than once because of an intervening write operation. The
+	// default value is false.
```
[the remark this answers is not in the document]
Comment (divjotarora) [reply mongo/options/findoptions.go:369]: done.

File: mongo/cursor.go (hunk 2 above)
Comment (divjotarora) [reply mongo/cursor.go:274]: done.

File: mongo/index_options_builder.go (hunk 1 above)
Comment (divjotarora) [reply mongo/index_options_builder.go:51]: done.

File: mongo/cursor.go [hunk 4]
```diff
@@ -271,8 +271,8 @@ func (c *Cursor) closeImplicitSession() {
 // BatchCursorFromCursor returns a driver.BatchCursor for the given Cursor. If there is no underlying
 // driver.BatchCursor, nil is returned.
 //
-// Deprecated: This is an unstable function because the driver.BatchCursor type exists in the "x" package. Both this
-// function and the driver.BatchCursor type should not be used by applications and may be changed or removed in any
+// Deprecated: This is an unstable function because the driver.BatchCursor type exists in the "x" package. Neither this
+// function nor the driver.BatchCursor type should not be used by applications and may be changed or removed in any
```
Comment (iwysiu) [inline mongo/cursor.go:275]: "should be"

Comment (divjotarora) [reply mongo/cursor.go:275]: done.

**Review by iwysiu — APPROVED** [review]

File: mongo/options/findoptions.go [hunk 5]
```diff
@@ -87,8 +89,10 @@ type FindOptions struct {
 	// The number of documents to skip before adding documents to the result. The default value is 0.
 	Skip *int64
 
-	// If true, the cursor will not return a document more than once because of an intervening write operation. This
-	// option has been deprecated in MongoDB version 4.0. The default value is false.
+	// If true, the cursor will not return a document more than once because of an intervening write operation. The
+	// default value is false.
+	//
+	// Deprecated: This option has been deprecated in MongoDB version 4.0.
```
Comment (kevinAlbs) [inline mongo/options/findoptions.go:95]:
```suggestion
	// Deprecated: This option has been deprecated in MongoDB version 3.6 and removed in MongoDB version 4.0.
```

Comment (kevinAlbs) [reply mongo/options/findoptions.go:95]: According to https://jira.mongodb.org/browse/DRIVERS-454

File: mongo/options/findoptions.go [hunk 6]
```diff
@@ -209,6 +215,8 @@ func (f *FindOptions) SetSkip(i int64) *FindOptions {
 }
 
 // SetSnapshot sets the value for the Snapshot field.
+//
+// Deprecated: This option has been deprecated in MongoDB version 4.0. The default value is false.
```
Comment (kevinAlbs) [inline mongo/options/findoptions.go:219]:
```suggestion
// Deprecated: This option has been deprecated in MongoDB version 3.6. The default value is false.
```

**Review by kevinAlbs — APPROVED** [review]
LGTM pending a tweak to the note about which server version snapshot was deprecated.

File: mongo/options/findoptions.go (hunk 5 above)
Comment (divjotarora) [reply mongo/options/findoptions.go]: Changed for both `FindOptions` and `FindOneOptions`.

File: mongo/options/findoptions.go (hunk 6 above)
Comment (divjotarora) [reply mongo/options/findoptions.go]: I wanted to keep the notice the same for the struct field and the setter so I made them both
```
// Deprecated: This option has been deprecated in MongoDB version 3.6 and removed in MongoDB version 4.0.
```

**Review by kevinAlbs — APPROVED** [review]
Latest commit LGTM

**Pull request merged**